pkg/config/config_test.go

```diff
@@ -12,19 +12,156 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+func TestUsingJustTokenEnvVar(t *testing.T) {
+	os.Clearenv()
+	viper.Reset()
+
+	viper.SetConfigFile(os.DevNull)
+	viper.Set(constants.ArgConfig, os.DevNull)
+
+	assert.NoError(t, os.Setenv(sdk.IonosTokenEnvVar, "tok"))
+	assert.NoError(t, Load())
+	assert.Equal(t, "tok", viper.GetString(Token))
+	assert.Equal(t, "", viper.GetString(Username))
+	assert.Equal(t, "", viper.GetString(Password))
+	assert.Equal(t, "", viper.GetString(ServerUrl))
+}
+
+func TestTokEnvWithUserPassConfigBackup(t *testing.T) {
+	// Useful for API routes which don't accept bearer tokens, or custom IonosCTL commands (Image Upload)
+	os.Clearenv()
+	viper.Reset()
+
+	assert.NoError(t, os.Setenv(sdk.IonosTokenEnvVar, "tok"))
+	path := filepath.Join("..", "testdata", "config_user_pass.json") // TODO: These files should be created and then destroyed by the tests
+	viper.SetConfigFile(path)
+	viper.Set(constants.ArgConfig, path)
+	assert.NoError(t, os.Chmod(path, 0600))
+	assert.NoError(t, Load())
+
+	assert.Equal(t, "tok", viper.GetString(Token))
+	assert.Equal(t, "test@ionos.com", viper.GetString(Username))
+	assert.Equal(t, "test", viper.GetString(Password))
+	assert.Equal(t, "", viper.GetString(ServerUrl))
+}
+
+func TestTokEnvWithFullConfig(t *testing.T) {
+	// Config token should not override env var token
+	os.Clearenv()
+	viper.Reset()
+
+	assert.NoError(t, os.Setenv(sdk.IonosTokenEnvVar, "tok"))
+	path := filepath.Join("..", "testdata", "config.json") // TODO: These files should be created and then destroyed by the tests
+	viper.SetConfigFile(path)
+	viper.Set(constants.ArgConfig, path)
+	assert.NoError(t, os.Chmod(path, 0600))
```

Comment:
@avirtopeanu-ionos not a big deal, you can leave it as it is, but since now you are only reading from the file, the number can be `0400`. You can leave it as it is since in the future you will create the test file, write some info in it, use the test file for the test and then delete the test file.

Reply:
there had been a discussion previouslyabout the config file permissions. The idea is that LoadFile() fails if the perms aren't exactly `0600`. I can't comment on why this is exactly, but this is the reason why we set exactly this perm number